### DISCUSSION TITLE
Upgrade docusaurus to v2 and add mobxjs/zh.mobx.js.org ?

### DISCUSSION THREAD

**Discussion by iChenLei** [mobxjs/mobx#2781] [Show and tell]
### Upgrade docusaurus to v2 ?
![image](https://user-images.githubusercontent.com/14012511/106897160-86f00c00-672d-11eb-95d0-b3ad3e4ca69f.png)

Recently I use docusaurus v2 to process our mobxjs/docs. 👆 Look this

#### The Good
* MDX support
* Dark mode support
* i18n support
* version doc support
* sync with docusaurus community
* and more

#### The bad
* Need change doc content to make `MDX` compatiable
* Need write some custom css to redecorate doc UI
* Not everyone like the new UX
* and more

### Create a repo for chinese(simplified) ?
Official doc's translate link is outdate ! and no one to maintain it.

![2021-02-04 21 22 28](https://user-images.githubusercontent.com/14012511/106898399-1813b280-672f-11eb-8bd3-8f8536b1f76f.png)

The latest mobx version is v6, and you know even  6.0.x and 6.1.x has big difference !
New repo `mobxjs/zh.mobx.js.org` for seek more volunteers to translate and help mobx community growth .

### Summary
@mweststrate @FredyC Any suggestion?

**Comment by danielkcz** [comment]
Well, it almost seems to me that benefits are not that great compared to work needed. However, if you have free time to spare, what's holding you back to start working on it?

It does also seem you are the only one who speaks Chinese around here? Unless you can find other volunteers, I don't think anyone here can help you with that really. I am not entirely sure why you want a separate repo though. What does i18n support mean with V2? Can we keep the translation in the monorepo?

**Comment by mweststrate** [comment]
Last time I checked only docusaurus v1 supports multiple languages, and not
all plugins / settings we are using are available in v2. Not sure if that
is still the case. But at least v1 is still the official version, and v2
beta, so not sure what we would benefit from v2?

**Comment by iChenLei** [comment]
## Q: Upgrade docusaurus to v2 ?

A: Ok, I get the conclusion that we don't need upgrade docusaurus to v2(at least now).

## Q: Can we add mobxjs/zh.mobx.js.org ?
> It does also seem you are the only one who speaks Chinese around here? 

Yes, it looks like that I am the only who speaks Chinese around here. but in china mobx is very popluar in IT company. At the company I work for，mobx is the default React state management library.(😃   need support low version browser so we use mobx4. chinese translatation is not fake needs.

> Unless you can find other volunteers, I don't think anyone here can help you with that really. 

I seek volunteers in `wechat`, `zhihu.com` and more cn platform, not in github discussions.

> What does i18n support mean with V2?

Sorry, I noticed that v1 also support i18n.

> I am not entirely sure why you want a separate repo though.  Can we keep the translation in the monorepo?

`vuejs/cn.vuejs.org` and `vuejs/jp.vuejs.org` are greate example. separate repo can avoid pollution(you can imagine that main monorepo filled with chinese PR and issues). It make maintainers focus on mobx self and not boring translation pr(or issue). And use separate repo can make us config fast pull request flow to update doc content. volunteers don't need clone project to local, just finish all work on github webpage.(edit(on webpage) -> create pr(on webpage) -> ci(create preview) -> merge(on webpage) -> ci(create production doc)  )

![2021-02-05 10 33 08](https://user-images.githubusercontent.com/14012511/106981748-8c853a80-679d-11eb-975a-f1661558b64c.png)

Why not just create a `iChenLei/zh.mobx.js.org` ? Translate is a boring and hard work. 

`https://github.com/suprise/mobx-cn`  latest update 4 years ago
`https://github.com/SangKa/MobX-Docs-CN`  latest update 2 years ago

`mobxjs/zh.mobx.js.org` can get high exposure , volunteers work for mobx official not personal repo. It make sense.

I can make mobx sticker, Giving volunteers as gift。`Mobx6 cheatsheet` paper also can be gift, and we can add volunteers list on readme to motivate volunteers.(It's not appropriate in mobx main monorepo.
![2021-02-05 11 01 42](https://user-images.githubusercontent.com/14012511/106983884-9610a180-67a1-11eb-89fa-ea3565e837c4.png)

 
### No one care these translate PR, it's too bad. 😭 

![2021-02-05 10 56 40](https://user-images.githubusercontent.com/14012511/106983419-d4f22780-67a0-11eb-84f1-f6aeefdf5af5.png)
![2021-02-05 10 57 24](https://user-images.githubusercontent.com/14012511/106983482-ef2c0580-67a0-11eb-9b0b-96e81a17b5c3.png)

Even in 2020, we also can found translate pull request ( `SangKa/MobX-Docs-CN` last update 2 years ago

**Reply by mweststrate** [reply]
> Ok, I get the conclusion that we don't need upgrade docusaurus to v2(at least now).

Sorry, for sounding to skeptic, that is not what I wanted to imply at all :). I was just wondering in general about benefits. I got a partial answer to that [here](https://github.com/immerjs/immer/pull/746#issuecomment-773739122) :)  Awesome for picking looking into this! For the Chinese translation we can definitely share stickers or cheatsheets, that sounds like a great idea! If you need to create stickers feel free to submit reasonable expenses to the opencollective!

**Comment by danielkcz** [comment]
Ok, I've created `mobxjs/zh.mobx.js.org` along with a separate team with admin access to that repo. @iChenLei you have been invited to that team. I am not sure if you can add more members to the team though, I hope so.

**Reply by iChenLei** [reply]
Thank you, sir. I will migrate official documents to this repo[zh.mobx.js.org], and add netlify auto deploy CI/CD as soon as possible. Of course, I will create a pr to `js.org` to register `zh.mobx.js.org` sub domain. 

>  I am not sure if you add can more members to the team though, I hope so.

It looks like only I need do translate work, so don't need care this. 🚀
